### DISCUSSION TITLE
Prisma Sveltekit Generated Client -> exports is not defined

### DISCUSSION THREAD

**Discussion by TheBigMemph** [prisma/prisma#21561] [Q&A] [answered]
### Question

i have two schema files in my sveltekit project (sqlServer, MongoDb) and i generated two clients with those schemas.

I setup this minimal reproduction here: https://github.com/TheBigMemph/test

If i load either one of the clients declared in lib/index.ts into the app the following error comes up:

```
12:22:05 [vite] Error when evaluating SSR module /prisma/generated/sql/index-browser.js:
|- ReferenceError: exports is not defined
    at /Users/***/***/test/my-app/prisma/generated/sql/index-browser.js:2:23
    at instantiateModule (file:///Users/***/***/test/my-app/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:56052:15)

12:22:05 [vite] Error when evaluating SSR module /src/lib/index.ts: failed to import "/prisma/generated/sql/index-browser.js"
|- ReferenceError: exports is not defined
    at /Users/***/***/test/my-app/prisma/generated/sql/index-browser.js:2:23
    at instantiateModule (file:///Users/***/***/test/my-app/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:56052:15)

12:22:05 [vite] Error when evaluating SSR module /src/routes/+page.ts: failed to import "/src/lib/index.ts"
|- ReferenceError: exports is not defined
    at /Users/***/***/test/my-app/prisma/generated/sql/index-browser.js:2:23
    at instantiateModule (file:///Users/***/***/test/my-app/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:56052:15)

ReferenceError: exports is not defined
    at /Users/***/***/test/my-app/prisma/generated/sql/index-browser.js:2:23
    at instantiateModule (file:///Users/***/***/test/my-app/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:56052:15)
ReferenceError: exports is not defined
    at /Users/***/***/test/my-app/prisma/generated/sql/index-browser.js:2:23
    at instantiateModule (file:///Users/***/***/test/my-app/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:56052:15)
```

### How to reproduce (optional)

https://github.com/TheBigMemph/test

### Information about Prisma Schema, Client Queries and Environment (optional)

https://github.com/TheBigMemph/test

**Comment by ludralph** [comment, marked as answer]
Hi @TheBigMemph 👋 

Can you please ensure that the Prisma client is generated inside the node_modules directory? This can be done by specifying the output path in your `schema.prisma` file to point to a directory within node_modules. For example:
```prisma
generator client {
  provider = "prisma-client-js"
  output   = "../node_modules/@prisma-app/client"
}
```
And then you would import your Prisma client like so:

```js
import { PrismaClient } from '@prisma-app/client';
```
This approach was mentioned in this [discussion](https://github.com/prisma/prisma/issues/2443), where users found that placing the generated client in the node_modules directory resolved the exports is not defined error.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by andrewdavid48** [reply]
That though creates another issue, this time with wasm.
```
ENOENT: no such file or directory, open '/node_modules/@prisma-app/client/query_engine_bg.wasm?module'

```
